protocol/03_encoding.md

```diff
@@ -25,8 +28,17 @@ The information here is used to receive the data JSON.
 |--------|---------------------------------------------------------|---------------------|
 | Length | Length of the 2nd JSON string (Data) in bytes.          | `uint16` or `int32` | 
 | CRC32  | CRC32 checksum over the buffer of the 2nd JSON string.  | `uint32` or `int64` |
+| type_id| The type of the message.                                | `uint8` or `int16`  |
+
+To ensure that the header is always 64 characters long, all values are encoded as strings with leading zeros. The number of digits depends on the largest possible number of these values. Both values must be treated as unsigned integers. If the chosen programming language does not support this, a larger data type must be selected. The recommended types are listed in the table above.
+
+### Packet Type
+The `type_id` field is used to determine the type of the message. The following types are defined:
 
-To ensure that the header is always 50 characters long, all values are encoded as strings with leading zeros. The number of digits depends on the largest possible number of these values. Both values must be treated as unsigned integers. If the chosen programming language does not support this, a larger data type must be selected. The recommended types are listed in the table above.
+| ID | Type          | Description                |
```

Comment:
We could add `3` for the case a message contains both, routing and routed. This also allows checking for bits after integer conversion.